### DISCUSSION TITLE
Enhanced OCR Options for a More Flexible Experience

### DISCUSSION THREAD

**Discussion by djblind** [nvaccess/nvda#15738] [Ideas]
Current Issue:
In the current version of NVDA, the use of the OCR (Optical Character Recognition) feature can be improved to provide a more flexible experience for users. We have identified three potential enhancements.
Proposals:
1.
Configurable OCR Frequency in Tenths of a Second:
Currently, OCR runs periodically at a predefined interval. We suggest adding the option for users to customize the OCR frequency by specifying an interval in tenths of a second. This customization would offer better adaptability, allowing users to adjust OCR based on their reading speed and specific needs.
2.
Selective OCR for a Defined Screen Area by Coordinates:
The current OCR feature analyzes the entire screen, which may not always be the most efficient way to obtain information. We propose adding the ability for users to define coordinates for a specific screen area to analyze. For example, a user could set coordinates to target a particular table by specifying "from x200 to x400 and from y100 to y250." This would enable users to precisely target the information they need, improving efficiency and accessibility.
3.
Option for Direct Vocal Feedback or Virtual Window for OCR, Ideal with Periodic OCR:
We suggest adding an option that allows users to choose between receiving direct vocal feedback of OCR results in the main application window or opening a virtual window, based on their specific needs. This feature would provide greater flexibility, especially when using periodic OCR.
Benefits:
•
Advanced customization: These features would provide users with more precise control over how they interact with on-screen content.
•
Efficiency: By reducing the time intervals between OCR recognition, targeting specific screen areas, and offering a direct vocal feedback option, users could access important information more quickly.
Conclusion:
The addition of these features would offer enhanced customization for NVDA's OCR function and significantly improve efficiency and accessibility for blind or visually impaired users. For instance, envision a gamer who could define a screen area to monitor real-time points of interest within a game, or a user watching a streaming video with subtitles who would automatically receive text descriptions of dialogues without disrupting their viewing experience.
We appreciate your consideration of these suggestions for future versions of NVDA.

**Comment by Adriani90** [comment]
cc: @michaelDCurran

**Comment by CyrilleB79** [comment]
Just to clarify, the OCR is not done on the whole screen but on the current navigator object's rectangle, which may or may not be the whole desktop, and which is not generally.
